### PR TITLE
openiscsi: 2.1.4 -> 2.1.7

### PR DESCRIPTION
**Pull request by trofi**
changelog: https://github.com/open-iscsi/open-iscsi/blob/master/Changelog
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by trofi** [comment]
Result of `nixpkgs-review pr 177804` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>openiscsi</li>
  </ul>
</details>

**Comment by trofi** [comment]
Does not actually fix the failure. I tested against merge base that did not have new linux-headers yet. To be sorted upstream.

**Comment by trofi** [comment]
Withdrawing the claim it fixes `staging-next`. We'll need to fix linuxheaders instead. But the version update should be fine. Retargeted against `master.

**Review by zaninime — APPROVED** [review]

File: pkgs/os-specific/linux/open-iscsi/default.nix [hunk 1]
```diff
@@ -23,8 +23,17 @@ stdenv.mkDerivation rec {
 
   preConfigure = ''
     sed -i 's|/usr|/|' Makefile
+
+    # Remove blanket -Werror. Fails for minor error on gcc-11.
+    sed -i 's| -Werror | |g' libopeniscsiusr/Makefile usr/Makefile
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/open-iscsi/default.nix:28]: We should use substituteInPlace to get a warning if it no longer matches.

Comment (trofi) [reply pkgs/os-specific/linux/open-iscsi/default.nix]: Oh, good point! I keep forgetting about it. Changed to `substituteInPlace usr/Makefile --replace ' -Werror ' ' '`

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**